docs/architecture.md

```diff
@@ -15,139 +15,207 @@ Prometheus instances scrape samples from various targets and then push them to C
 
 Cortex requires that each HTTP request bear a header specifying a tenant ID for the request. Request authentication and authorization are handled by an external reverse proxy.
 
-Incoming samples (writes from Prometheus) are handled by the [distributor](#distributor) while incoming reads (PromQL queries) are handled by the [query frontend](#query-frontend).
+Incoming samples (writes from Prometheus) are handled by the [distributor](#distributor) while incoming reads (PromQL queries) are handled by the [querier](#querier) or optionally by the [query frontend](#query-frontend).
+
+## Storage
+
+Cortex currently supports two storage engines to store and query the time series:
+
+- Chunks (default, stable)
+- Blocks (experimental)
+
+The two engines mostly share the same Cortex architecture with few differences outlined in the rest of the document.
+
+### Chunks storage (default)
+
+The chunks storage stores each single time series into a separate object called chunk. Each chunk contains the timestamp-value pairs for a given period (defaults to 12 hours) of the series. Series chunks are then indexed by time range period and metric labels, in order to provide a fast lookup across millions - or even billions - of chunks.
```

Comment:
```suggestion
The Chunks storage stores each single time series into a separate object called _Chunk_. Each Chunk contains the timestamp-value pairs for a given period (defaults to 12 hours) of the series. Chunks are then indexed by time range and labels, in order to provide a fast lookup across many (over millions) chunks.
```

Reply:
Sorry if this is a bit invasive, in this paragraph you're describing a Chunk and its purpose. Seems like in this case a Chunk becomes the subject of the paragraph.

[Prometheus describes](https://github.com/prometheus/prometheus/blob/master/tsdb/docs/format/chunks.md#chunk) a Chunk as a proper noun. The only case where it doesn't is as it refers to chunk files. I know these are not TSDB Chunks but I'm using that documentation as an example.

Reply:
Good point, thanks!